framework/decode/metadata_json_consumer.h

```diff
@@ -0,0 +1,275 @@
+        FieldToJson(jdata["height"], height, json_options);
+        FieldToJson(jdata["pre_transform"], pre_transform, json_options);
+        WriteBlockEnd();
+    }
+
+    virtual void
+    ProcessCreateHardwareBufferCommand(format::HandleId                                    memory_id,
+                                       uint64_t                                            buffer_id,
+                                       uint32_t                                            format,
+                                       uint32_t                                            width,
+                                       uint32_t                                            height,
+                                       uint32_t                                            stride,
+                                       uint64_t                                            usage,
+                                       uint32_t                                            layers,
+                                       const std::vector<format::HardwareBufferPlaneInfo>& plane_info) override
+    {
+        const util::JsonOptions& json_options = GetOptions();
+        auto&                    jdata        = WriteMetaCommandStart("CreateHardwareBufferCommand");
+        HandleToJson(jdata["memory_id"], memory_id, json_options);
+        HandleToJson(jdata["buffer_id"], buffer_id, json_options);
+        FieldToJson(jdata["format"], format, json_options);
+        FieldToJson(jdata["width"], width, json_options);
+        FieldToJson(jdata["height"], height, json_options);
+        FieldToJson(jdata["stride"], stride, json_options);
+        FieldToJson(jdata["usage"], usage, json_options);
+        FieldToJson(jdata["layers"], layers, json_options);
+        WriteBlockEnd();
+    }
+
+    virtual void ProcessDestroyHardwareBufferCommand(const uint64_t buffer_id) override
+    {
+        auto& jdata = WriteMetaCommandStart("DestroyHardwareBufferCommand");
+        HandleToJson(jdata["buffer_id"], buffer_id, GetOptions());
+        WriteBlockEnd();
+    }
+
+    virtual void ProcessSetDevicePropertiesCommand(format::HandleId   physical_device_id,
+                                                   uint32_t           api_version,
+                                                   uint32_t           driver_version,
+                                                   uint32_t           vendor_id,
+                                                   uint32_t           device_id,
+                                                   uint32_t           device_type,
+                                                   const uint8_t      pipeline_cache_uuid[format::kUuidSize],
+                                                   const std::string& device_name) override
+    {
+        const util::JsonOptions& json_options = GetOptions();
+        this->writer_->WriteMetaCommand("SetDevicePropertiesCommand", [&](auto& jdata) {
```

Comment:
Convert all the ` this->writer_->WriteMetaCommand(` forms using lambda in this class to the simpler form. used in some of the other functions here.